### ISSUE TITLE
Limit valid RDF type values

### ISSUE THREAD

**Issue by FynnBe** [bioimage-io/spec-bioimage-io#518]
TLDR; we don't need arbitrary RDF types, let's restrict them to:  model, application, notebook, dataset, collection, (coming soon: op) and generic (a new generic type for any resource that does not fit the other types)

So far we have talked about different RDF types, namely: model, application, notebook, dataset, collection, workflow (now known as 'op'). No other RDF type has been contributed so far and I think we should not allow for arbitrary types, the flexibility gained (that we haven't used) is not worth worrying about typos, IMO.
At least we can make an arbitrary RDF type explicit by introducing a 'generic' type.

On bioimage.io we have dedicated tabs for: model, application (includes notebook) and dataset.
We knowingly ignore the collection type for now and will adapt the GUI once workflows/ops are ready to be published.

Although this is a breaking change it really is not, as there are no RDF with other types out there + auto-convert can cast any unknown RDF type to 'generic' if someone somewhere does have a private RDF with a specific type.

how about some more breaking changes, @oeway ? 😆
cc @constantinpape @k-dominik @Tomaz-Vieira

**Comment by esgomezm** [comment]
Hey @FynnBe,

I fully agree with limiting the RDF file type. However, after interacting with students/users in the schools I have very strong concerns about how the information is displayed on the website and how we manage to properly communicate with them. (This has more to do with tabs' info rather than how we classify the resources in the infrastructure). 
For example, it is clear to us what workflows or operations mean and why they are not necessarily a notebook. However, you could contribute a notebook that runs a workflow or a pipeline. In reality, the user will not care what we call it, as they simply want to easily find and use these pipelines. 

I raised this issue in the meetings several times, and I would be happy to further discuss it whenever you want.

**Comment by oeway** [comment]
Hey, why don’t we implement this as a feature in the CI or spec library
only? I would leave the spec as is and perhaps introduce a strict type mode
in the spec validator which can be switched on for the CI.

For the spec itself, I don’t want to be the one always say no, but I still
felt like it’s a bit abrupt to introduce breaking changes just because of
“avoiding typos” if I understand correctly from your comments. The generic
RDF spec are designed to be stable and future proof, backward-compatible,
so we won’t need to upgrade the format version all the time. It’s good to
avoid typos but it should not be a feature that worth pursuing by
introducing breaking changes. IMO, we should simply correct the typo if
that happens, many other fields (e.g. config) can contain typos too and we
are not able to detect those from the spec.

For the types (BTW, we merged notebooks to applications, so there is no RDF
with notebook type), I don’t see the point of casting other types into
generic, that means a future new type need to use config to differentiate
their actual type — we are shifting the problem from one key to another.
Let’s say we decided to add a new type called “prompt” for LLM-guided
analysis, we will have to bump the format version and cause a lot of
unnecessary changes in the underlying infrastructure (the website, CI, and
also the core library which is installed locally and not always updated).

**Comment by oeway** [comment]
> Hey @FynnBe,
> 
> 
> 
> I fully agree with limiting the RDF file type. However, after interacting with students/users in the schools I have very strong concerns about how the information is displayed on the website and how we manage to properly communicate with them. (This has more to do with tabs' info rather than how we classify the resources in the infrastructure). 
> 
> For example, it is clear to us what workflows or operations mean and why they are not necessarily a notebook. However, you could contribute a notebook that runs a workflow or a pipeline. In reality, the user will not care what we call it, as they simply want to easily find and use these pipelines. 
> 
> 
> 
> I raised this issue in the meetings several times, and I would be happy to further discuss it whenever you want.

Hey Esti, are you proposing to improve the tab organization? Do you have a more concrete suggestion on how to improve that? Would be happy to chat more in details during the meeting.
